docs/book/component-guide/deployers/kubernetes.md

```diff
@@ -0,0 +1,433 @@
+---
+description: Deploy ZenML pipeline services on Kubernetes clusters.
+---
+
+# Kubernetes Deployer
+
+The Kubernetes deployer is a [deployer](./) flavor provided by the Kubernetes
+integration. It provisions ZenML pipeline deployments as long-running services
+inside a Kubernetes cluster.
+
+{% hint style="warning" %}
+This component is only meant to be used within the context of a [remote ZenML installation](https://docs.zenml.io/getting-started/deploying-zenml/). Usage with a local ZenML setup may lead to unexpected behavior!
+{% endhint %}
+
+## When to use it
+
+Reach for the Kubernetes deployer when you need:
+
+* you're already using Kubernetes.
+* **Production-grade serving** with managed Kubernetes clusters (EKS, GKE, AKS,
+  etc.) or self-hosted clusters.
+* **Multi-replica deployments** and load-balanced access to your pipeline
+  service.
+* **Fine-grained pod customization** (resources, tolerations, secrets,
+  affinity, custom command/args, probes).
+* **Cluster networking integrations** (e.g., exposing services via
+  LoadBalancer, NodePort, or internal ClusterIP plus your own Ingress/mesh).
+
+If you only need local development or single-node deployments, consider the
+[local](local.md) or [docker](docker.md) deployer flavors instead.
+
+## How to deploy it
+
+1. Install the Kubernetes integration:
+
+   ```bash
+   zenml integration install kubernetes
+   ```
+
+2. Ensure your stack contains a container registry and an image builder that the
+   Kubernetes cluster can access. Remote clusters require remote registries.
+
+3. Configure access to your cluster. You can either:
+
+   - Provide a kubeconfig context via `kubernetes_context`.
+   - Run the client inside the cluster (`incluster=True`).
+   - Link a [Kubernetes service connector](../../service-connectors/connector-types/kubernetes-cluster.md).
+
+4. Register and activate the deployer:
+
+   ```bash
```

Comment:
I might be wrong, but I have a feeling that you can't indent these code blocks? (i.e. they won't render unless they're all the way to the left of the page?)